lib/resty/upstream/healthcheck.lua

```diff
@@ -661,22 +673,102 @@ local function gen_peers_status_info(peers, bits, idx)
         if peer.down then
             bits[idx + 2] = " DOWN\n"
         else
-            bits[idx + 2] = " up\n"
+            bits[idx + 2] = " UP\n"
         end
         idx = idx + 3
     end
     return idx
 end
 
+local function new_status_table(n)
+    local new_status_meta = {
+        __add = function(self, rhs)
```

Comment:
move the function outside, so we do not create the lua closure every healthy check.
The lua closure can not be jit compiled

Reply:
the whole meta table can be move outside this function.

Reply:
Will do, good idea